### ISSUE TITLE
Getting error when trying to enable coscheduler following develop instructions and tutorial

### ISSUE THREAD

**Issue by Frans-Lukas** [kubernetes-sigs/scheduler-plugins#150]
Moved from https://github.com/kubernetes/website/issues/21128,

Using minikube, I compiled the scheduler-plugins image (according to [these instructions](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/doc/develop.md)), and set the image in my `/etc/kubernetes/manifests/kube-scheduler.yaml` file of my master node. I.e:
```
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  labels:
    component: kube-scheduler
    tier: control-plane
  name: kube-scheduler
  namespace: kube-system
spec:
  containers:
  - command:
    - kube-scheduler
    - --authentication-kubeconfig=/etc/kubernetes/scheduler.conf
    - --authorization-kubeconfig=/etc/kubernetes/scheduler.conf
    - --config=/etc/kubernetes/coscheduling-config.yaml
    - --bind-address=127.0.0.1
    - --kubeconfig=/etc/kubernetes/scheduler.conf
    - --leader-elect=false
    - --port=0
    image: localhost:5000/scheduler-plugins/kube-scheduler:latest   <---- The compiled custom scheduler
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 8
 .....
```
Which I thought would create the scheduler pod with the custom scheduler image but I get the error message:
```
W0219 17:10:50.073294       1 client_config.go:613] error creating inClusterConfig, falling back to default config: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
couldn't create scheduler using provider "DefaultProvider": initializing profiles: creating profile for scheduler name default-scheduler: error initializing plugin "Coscheduling": failed to init rest.Config: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```

@Huang-Wei

**Comment by xq2005** [comment]
@Frans-Lukas Could you paste `volumeMounts`, `volumes` detail and `/etc/kubernetes/coscheduling-config.yaml` content?

**Comment by Frans-Lukas** [comment]
`volumeMounts` and `volumes` of `/etc/kubernetes/manifests/kube-scheduler.yaml`:
```
...
    volumeMounts:
    - mountPath: /etc/kubernetes/scheduler.conf
      name: kubeconfig
      readOnly: true
    - mountPath: /etc/kubernetes/coscheduler-config.yaml
      name: config
      readOnly: true
  hostNetwork: true
  priorityClassName: system-node-critical
  volumes:
  - hostPath:
      path: /etc/kubernetes/scheduler.conf
      type: FileOrCreate
    name: kubeconfig
  - hostPath:
      path: /etc/kubernetes/coscheduler-config.yaml
      type: FileOrCreate
    name: config
status: {}
```
`/etc/kubernetes/coscheduling-config.yaml`:

```
apiVersion: kubescheduler.config.k8s.io/v1beta1
kind: KubeSchedulerConfiguration
clientConnection:
  kubeconfig: /etc/kubernetes/scheduler.conf
profiles:
- schedulerName: default-scheduler
  plugins:
    queueSort:
      enabled:
        - name: Coscheduling
      disabled:
        - name: "*"
    preFilter:
      enabled:
        - name: Coscheduling
    permit:
      enabled:
        - name: Coscheduling
    reserve:
      enabled:
        - name: Coscheduling
    postBind:
      enabled:
        - name: Coscheduling
```

**Comment by damemi** [comment]
`invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable` seems like possibly an issue with the kubeconfig. Is there a file `/etc/kubernetes/scheduler.conf` on the host?

**Comment by xq2005** [comment]
pluginConfig is required in /etc/kubernetes/coscheduling-config.yaml

https://github.com/kubernetes-sigs/scheduler-plugins/blob/ff6de60898d9d6867a94b37383ba641a6a7e0f0c/manifests/coscheduling/scheduler-config.yaml#L26-L34

**Comment by Frans-Lukas** [comment]
@damemi `/etc/kubernetes/scheduler.conf` exists and contains: 

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: 
...
  server: https://...
  name: kind
contexts:
- context:
    cluster: kind
    user: system:kube-scheduler
  name: system:kube-scheduler@kind
current-context: system:kube-scheduler@kind
kind: Config
preferences: {}
users:
- name: system:kube-scheduler
  user:
...
```

@xq2005 I added the pluginConfig to my `/etc/kubernetes/coscheduling-config.yaml` but I do not understand what `kubeMaster` is? 

Thanks!

**Comment by xq2005** [comment]
@Frans-Lukas Only adding kubeConfigPath in the pluginConfig is enough, you can ignore kubeMaster and have a try.

**Comment by Huang-Wei** [comment]
@Frans-Lukas As @xq2005 mentioned, setting `kubeConfigPath` as the args of coscheduing is a must. It's because the scheduler framework doesn't expose its original kubeConfig credentials, and hence the plugin has to provide it again; otherwise the plugin cannot initialize its CRD informer.

It's really a pain point, hope we can come out with a secure way to resolve it in the near future.

Ref #98, https://github.com/kubernetes/kubernetes/issues/96427

---

@xq2005 could you please update the install doc to emphasize setting `kubeConfigPath`? Thanks in advance.

**Comment by Frans-Lukas** [comment]
@Huang-Wei @xq2005 
After adding 
```
  pluginConfig:
    - name: Coscheduling
      args:
        permitWaitingTimeSeconds: 10
        deniedPGExpirationTimeSeconds: 3
        kubeConfigPath: /etc/kubernetes/scheduler.conf

```
to our `coshceduler-config.yaml` file we are getting another error regarding what I think is a RBAC authentication error. 

```
E0225 09:18:44.749970       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
E0225 09:18:45.733099       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
E0225 09:18:48.245497       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
E0225 09:18:53.327969       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
E0225 09:19:05.902512       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
```

I tried looking up a resource called `podgroups`, but couldn't find anything that seemed to grant access to it.

**Comment by xq2005** [comment]
@Frans-Lukas podgroups is a CRD, it must be created with following yaml.

https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/manifests/coscheduling/crd.yaml

**Comment by xq2005** [comment]
> @xq2005 could you please update the install doc to emphasize setting kubeConfigPath? Thanks in advance.

@Huang-Wei the install doc uses release v0.18.9, which is build by branch release-1.18. podgroups and pluginConfig are not included in branch release-1.18. 

I think we can enhance the develop.md. Adding a part about how to create crd and modify scheduler-config.yaml for each plugin.

**Comment by Frans-Lukas** [comment]
It definitely feels like we are getting closer but now we are still getting authentication errors after trying to create cluster roles.

`
E0226 10:25:04.725393       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
`

We tried creating a role with `kubectl create -f podgroup-reader.yaml` and could list it with `kubectl get clusterroles`:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  # "namespace" omitted since ClusterRoles are not namespaced
  name: podgroup-reader
rules:
- apiGroups: [""]
  #
  # at the HTTP level, the name of the resource for accessing Secret
  # objects is "secrets"
  resources: ["podgroups"]
  verbs: ["get", "watch", "list"]
```

We bound it to the scheduler user and the kube-system namespace to be redundant:
```
kubectl create rolebinding serviceaccounts-reader-auth-podgroups \
  --clusterrole=podgroup-reader \
  --group=system:serviceaccounts:kube-system \
  --namespace=kube-system

kubectl create rolebinding scheduler-reader-auth-podgroups \
  --clusterrole=podgroup-reader \
  --serviceaccount=kube-system:kube-scheduler \
  --namespace=kube-system

```

But none of the rolebindings changed the output from the scheduler, i.e. it continued to log:
`
E0226 10:25:04.725393       1 reflector.go:127] sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.PodGroup: failed to list *v1alpha1.PodGroup: podgroups.scheduling.sigs.k8s.io is forbidden: User "system:kube-scheduler" cannot list resource "podgroups" in API group "scheduling.sigs.k8s.io" at the cluster scope
`

**Comment by xq2005** [comment]
podgroups' apiGroups is "scheduling.sigs.k8s.io"

**Comment by seanmalloy** [comment]
> > @xq2005 could you please update the install doc to emphasize setting kubeConfigPath? Thanks in advance.
> 
> @Huang-Wei the install doc uses release v0.18.9, which is build by branch release-1.18. podgroups and pluginConfig are not included in branch release-1.18.
> 
> I think we can enhance the develop.md. Adding a part about how to create crd and modify scheduler-config.yaml for each plugin.

@xq2005 I submitted https://github.com/kubernetes-sigs/scheduler-plugins/pull/160 to back port your awesome docs to the `release-1.18` branch. Would you be wiling to update the docs on the master branch to reflect the required changes for the latest code on master?

**Comment by Huang-Wei** [comment]
@Frans-Lukas the doc was written for version v0.18.9, and that version doesn't have a PodGroup CRD. We're now working on releasing v0.19.X, which introduces PodGroup CRD, and hence the doc needs to be updated - not only the wordings, but also like RBAC, ServiceAccount. I've locally walked everything through, stay tuned.

/assign
